src/main/main.ts

```diff
@@ -162,6 +179,60 @@ ipcMain.on(ChannelEnum.HEALTH_CHECK, async (event) => {
   }
 });
 
+ipcMain.on(ChannelEnum.LOAD_PRESET, async (event, arg) => {
+  const channel = ChannelEnum.LOAD_PRESET;
+  const presetName = arg[0];
+  console.log(`Loading preset: ${presetName}`);
+
+  // TODO: should we do some str checking here?
+  try {
+    const presetSettings: IPreset = fetchPreset(presetName);
+    state.preAmp = presetSettings.preAmp;
+    state.filters = presetSettings.filters;
+    await handleUpdate(event, channel);
+  } catch (ex) {
+    console.log('Failed to read preset: ', presetName);
+    throw ex;
```

Comment:
I'm not sure if I mentioned to throw exceptions but if I did, sorry thats not the right thing to do here. Instead, we should be calling handleError() and passing a corresponding return code. Add a corresponding message to errors.ts and then this should be good. See handleUpdateHelper for an example.